packages/server/api/routes/variants.router.js

```diff
@@ -0,0 +1,41 @@
+const express = require('express');
+
+const router = express.Router({ mergeParams: true });
+
+// controllers
+const variantsController = require('../controllers/variants.controller');
+
+/**
+ * @swagger
+ * /variants?product=:
+ *  get:
+ *    tags:
+ *    - variants
+ *    summary: Get variants by productId
+ *    description:
+ *      Will return All variants with a matching productId.
+ *    produces: application/json
+ *    parameters:
+ *     - in: query
+ *       name: product
+ *       schema:
+ *         type: integer
+ *         required: true
+ *         description: The ID of the variants to get
+ *
+ *    responses:
+ *      200:
+ *        description: Successful request
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.get('/', (req, res, next) => {
+  if ('product' in req.query) {
+    variantsController
+      .getVariantsByProductId(req.query.product)
+      .then((result) => res.json(result))
+      .catch(next);
+  }
```

Comment:
I know this PR is already merged but this is a pattern we should really avoid.

We should not create branches based on if a parameter is present or not, we should write our code as if all parameters were sent and instead provide good defaults. If the parameter is required but no good default can be found, the API should throw an error instead.